streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableImpl.java

```diff
@@ -398,18 +444,46 @@ public V apply(final K key, final Change<V> change) {
         }
 
         final KTableKTableJoinMerger<K, R> joinMerge = new KTableKTableJoinMerger<>(
-                new KTableImpl<K, V, R>(builder, joinThisName, joinThis, sourceNodes, this.queryableStoreName, false),
-                new KTableImpl<K, V1, R>(builder, joinOtherName, joinOther, ((KTableImpl<K, ?, ?>) other).sourceNodes,
-                        ((KTableImpl<K, ?, ?>) other).queryableStoreName, false),
-                internalQueryableName
+            new KTableImpl<K, V, R>(builder, joinThisName, joinThis, sourceNodes, this.queryableStoreName, false, null),
```

Comment:
These `KTable` implementations aren't used individually to construct the overall topology but are components of the join merger.

Reply:
Sorry, but cannot follow. Can you elaborate?

Reply:
Sorry, re-reading my original description I realize it is a bit unclear.

In most cases the `KStream` or `KTable` instance created is returned to the caller, thus becoming a parent node in the graph and is eventually used to build the topology.  In those cases the graphNode created is passed to the newly created `KStream` or `KTable` instance as the parent graph node for the next operation.

In this case,  the two `KTable` instances in question are components in the `KTableKTableJoinMerger` class and never have any child nodes, since they are used by the join merge processor.  There is no graphNode created to pass along as the parent graph node.

Does this make sense?

Reply:
At a high level yes -- still need to think about the details. Thanks.